src/lib/syncable_ledger/syncable_ledger.ml

```diff
@@ -340,7 +353,14 @@ end = struct
             Either.First
               (Num_accounts
                  (len, MT.get_inner_hash_at_addr_exn mt content_root_addr) )
+        | What_child_hashes a ->
+            let ledger_depth = MT.depth mt in
+            let addresses = intermediate_range ledger_depth a subtree_depth in
+            let get_hash a = MT.get_inner_hash_at_addr_exn mt a in
```

Comment:
When this hits the ledger DB, looping over these one at a time will be slow. You should use one of the batch functions instead.

Reply:
Which function are you thinking about?
I considered that but there doesn't seem to be functions for arbitrary batch reads. 
There is `get_all_accounts_rooted_at_exn` which provides subtrees but not with variable depth, and internally seems to be looping over a list and reading each entry individually.

Reply:
It seems that the underlying `Ledger.get_hash_batch` isn't exposed. I suppose this can wait for later then.